homeassistant/components/google_drive/backup.py

```diff
@@ -0,0 +1,142 @@
+"""Backup platform for the Google Drive integration."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncIterator, Callable, Coroutine
+import logging
+from typing import Any
+
+from google_drive_api.exceptions import GoogleDriveApiError
+
+from homeassistant.components.backup import AgentBackup, BackupAgent, BackupAgentError
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import ChunkAsyncStreamIterator
+
+from . import DATA_BACKUP_AGENT_LISTENERS, GoogleDriveConfigEntry
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_get_backup_agents(
+    hass: HomeAssistant,
+    **kwargs: Any,
+) -> list[BackupAgent]:
+    """Return a list of backup agents."""
+    entries = hass.config_entries.async_loaded_entries(DOMAIN)
+    return [GoogleDriveBackupAgent(entry) for entry in entries]
+
+
+@callback
+def async_register_backup_agents_listener(
+    hass: HomeAssistant,
+    *,
+    listener: Callable[[], None],
+    **kwargs: Any,
+) -> Callable[[], None]:
+    """Register a listener to be called when agents are added or removed.
+
+    :return: A function to unregister the listener.
+    """
+    hass.data.setdefault(DATA_BACKUP_AGENT_LISTENERS, []).append(listener)
+
+    @callback
+    def remove_listener() -> None:
+        """Remove the listener."""
+        hass.data[DATA_BACKUP_AGENT_LISTENERS].remove(listener)
+
+    return remove_listener
+
+
+class GoogleDriveBackupAgent(BackupAgent):
+    """Google Drive backup agent."""
+
+    domain = DOMAIN
+
+    def __init__(self, config_entry: GoogleDriveConfigEntry) -> None:
+        """Initialize the cloud backup sync agent."""
+        super().__init__()
+        self.name = config_entry.title
```

Comment:
The name is used to derive the agent id, we need something stable, not the config entry title which can be changed by user.
We should consider separating between a stable id and a user visible name to `BackupAgent`. We may want to wait with merge of this until that has been done.

Reply:
I changed it to unique_id which is the email address